.github/workflows/scripts/linux_arm64/build_catalyst.sh

```diff
@@ -0,0 +1,81 @@
+set -e -x
+cd /catalyst
+git config --global --add safe.directory '*'
+# Process args
+export GCC_VERSION=$1
+export PYTHON_VERSION=$2
+export PYTHON_SUBVERSION=$3
+export PYTHON_PACKAGE_NAME=$4
+# Install system dependencies
+dnf update -y 
+dnf install -y libzstd-devel gcc-toolset-${GCC_VERSION} ${PYTHON_PACKAGE_NAME} ${PYTHON_PACKAGE_NAME}-devel
+dnf clean all -y 
+# Make GCC the default compiler
+source /opt/rh/gcc-toolset-${GCC_VERSION}/enable -y 
+export C_COMPILER=/opt/rh/gcc-toolset-${GCC_VERSION}/root/usr/bin/gcc 
+export CXX_COMPILER=/opt/rh/gcc-toolset-${GCC_VERSION}/root/usr/bin/g++ 
+# Set the right Python interpreter
+rm -rf /usr/bin/python3
+ln -s /opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/bin/python3 /usr/bin/python3 
+export PYTHON=/usr/bin/python3
+# Add LLVM, Python and GCC to the PATH env var
+export PATH=/catalyst/llvm-build/bin:/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/bin:/opt/rh/gcc-toolset-${GCC_VERSION}/root/usr/bin:$PATH
+# Install python dependencies
+/usr/bin/python3 -m pip install numpy pybind11 PyYAML cmake ninja delocate 'amazon-braket-pennylane-plugin>1.27.1'
+# Build Catalyst runtime
+cmake -S runtime -B runtime-build -G Ninja \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_LIBRARY_OUTPUT_DIRECTORY=runtime-build/lib \
+    -DCMAKE_CXX_STANDARD_LIBRARIES="-lpython${PYTHON_VERSION}" \
+    -DPYTHON_EXECUTABLE=/usr/bin/python3 \
+    -DPYTHON_INCLUDE_DIR=/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/include/python${PYTHON_VERSION} \
+    -DPYTHON_LIBRARY=/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/lib \
+    -Dpybind11_DIR=/opt/_internal/cpython-${PYTHON_VERSION}.${PYTHON_SUBVERSION}/lib/python${PYTHON_VERSION}/site-packages/pybind11/share/cmake/pybind11 \
+    -DENABLE_LIGHTNING_KOKKOS=ON \
+    -DLIGHTNING_GIT_TAG=latest_release \
+    -DENABLE_LAPACK=OFF \
+    -DKokkos_ENABLE_SERIAL=ON \
+    -DKokkos_ENABLE_OPENMP=ON \
+    -DKokkos_ENABLE_COMPLEX_ALIGN=OFF \
+    -DENABLE_WARNINGS=OFF \
+    -DENABLE_OPENQASM=ON \
+    -DENABLE_OPENMP=OFF \
+    -DLQ_ENABLE_KERNEL_OMP=OFF
+cmake --build runtime-build --target rt_capi rtd_lightning rtd_openqasm rtd_dummy
+# Patch OQC for correct linking
+echo "target_link_directories(rtd_oqc PRIVATE /catalyst/runtime-build/utils/runtime-build/lib)" >> frontend/catalyst/third_party/oqc/src/CMakeLists.txt
```

Comment:
Rather than writing and modifying this file, is it possible to instead make the dependent library location accessible within CMake and/or the path be provided as an argument?

Reply:
No idea how

Reply:
You can set a variable (e.g.  https://cmake.org/cmake/help/latest/command/set.html) in https://github.com/PennyLaneAI/catalyst/blob/main/frontend/catalyst/third_party/oqc/src/CMakeLists.txt that if not set, does nothing, but accepts the library dir you want to provide, and adds it if needed. Something like (copied from the link and adjusted):

```cmake
set_target_properties(rtd_oqc PROPERTIES BUILD_RPATH "$ORIGIN/../utils")
target_link_directories(rtd_oqc PRIVATE ${runtime_lib})
target_link_libraries(rtd_oqc PRIVATE pybind11::module catalyst_python_interpreter)

### New
if(${OQC_DEV_LIB_DIR})
    target_link_directories(rtd_oqc PRIVATE  "${OQC_DEV_LIB_DIR}")
endif()
###

set_property(TARGET rtd_oqc PROPERTY POSITION_INDEPENDENT_CODE ON)
file(COPY ${PROJECT_SOURCE_DIR}/oqc.toml DESTINATION ./backend)

add_subdirectory(tests)
```

This will then avoid creating files, as the cmake command can pass this with `cmake ..... -DOQC_DEV_LIB_DIR="...."`, or similar.

Reply:
Done